File: compute_lyapunov.py
import numpy as np
from lyap_maps import henon, henon_jacobian

def get_L(x0,y0, function, iterates=450):
    xs = [x0]
    ys = [y0]
    f_primes = []
    for i in range(iterates):
        new_x, new_y = function(xs[i], ys[i])
        xs.append(new_x)
        ys.append(new_y)

        dx = xs[i+1]-xs[i]
        dy = ys[i+1]-ys[i]
        dydx = np.gradient(dy, dx)
        f_primes.append(abs(dydx))
        # TODO: Fix this so that is actually finding the L per partial differential
    L_num = np.prod(f_primes)
    L_exp = np.log(L_num**(1/iterates))

    return L_exp

File: compute_lyapunov.jl
map(x) = 2.5*x*(1-x)
map¹(x) = 2.5 - 5*x

function get_L(x₀, iterates=450)
    xs = [x₀]
    mag_slopes = []
    for i=1:iterates
        append!(xs, map(xs[i]))
        append!(mag_slopes, abs(map¹(xs[i])))
    end
    return xs, mag_slopes
end

x₀s = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
L_exp = []
for x₀ in x₀s
    xs, mags = get_L(x₀)
    L_num = prod(mags)^(1/450)
    append!(L_exp, log(L_num))
end
println(L_exp)